library/system/src/lib/yast2/system_service.rb

```diff
@@ -136,8 +164,57 @@ def start_mode=(mode)
     def start_modes
       return @start_modes if @start_modes
       @start_modes = [:on_boot, :manual]
-      @start_modes << :on_demand if socket?
+      @start_modes << :on_demand if socket
       @start_modes
     end
+
+  private
+
+    # @return [Hash<String,Object>]
+    attr_reader :changes
+
+    # Get the current start_mode
+    def current_start_mode
+      @current_start_mode ||=
+        if service.enabled?
+          :on_boot
+        elsif socket && socket.enabled?
+          :on_demand
+        else
+          :manual
+        end
+    end
+
+    # Sets start mode to the underlying system
+    def save_start_mode
+      return unless changes[:start_mode]
+      case changes[:start_mode]
+      when :on_boot
+        service.enable
+        socket.disable
+      when :on_demand
+        service.disable
+        socket.enable
+      when :manual
+        service.disable
+        socket.disable
+      end
+    end
+
+    # Sets service status
+    def set_current_status
+      if changes[:active] && !service.active?
```

Comment:
Should the socket be started?

Reply:
Good point. I would say 'yes', but in this case the "activate" is about having the service running (the "Start/Stop" button). I will add a note in the PBI to think/discuss more about it.